pkg/client/client_venafi_cloud.go

```diff
@@ -260,13 +261,20 @@ func (c *VenafiCloudClient) Post(path string, body io.Reader) (*http.Response, e
 		return nil, err
 	}
 
-	req, err := http.NewRequest(http.MethodPost, fullURL(c.baseURL, path), body)
+	encodedBody := &bytes.Buffer{}
+	gz := gzip.NewWriter(encodedBody)
+	if _, err := io.Copy(gz, body); err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, fullURL(c.baseURL, path), encodedBody)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Accept", "application/json")
 	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Content-Encoding", "gzip")
```

Comment:
Is there a reason this header is missing in the example you show in the PR description?

Reply:
Well caught, I hadn't noticed that.

No idea why it doesn't show. This seems like a problem, I'll double check.

Reply:
I've added four unit tests to make sure this is not a problem anymore:

- a unit test checks that the headers are correctly set in Key Pair mode..
- a unit test checks that the headers are correctly set in Venafi Connection mode..
- a unit test checks that no compression takes place when `--disable-compression` is used in Key Pair mode.
- a unit test checks that no compression takes place when `--disable-compression` is used in Venafi Connection mode.